drivers/flash/flash_npcx_fiu_nor.c

```diff
@@ -284,20 +285,22 @@ static int flash_npcx_nor_erase(const struct device *dev, off_t addr,
 		/* Send chip erase command */
 		flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_CE);
 		return flash_npcx_nor_wait_until_ready(dev);
-	} else if (config->min_erase_size == KB(4)) {
-		opcode = SPI_NOR_CMD_SE;
-	} else if (config->min_erase_size == KB(64)) {
-		opcode = SPI_NOR_CMD_BE;
-	} else {
-		return -EINVAL;
 	}
 
 	while (size > 0) {
 		flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_WREN);
 		/* Send page/block erase command with addr */
-		flash_npcx_uma_cmd_by_addr(dev, opcode, addr);
-		addr += config->min_erase_size;
-		size -= config->min_erase_size;
+		if ((size >= BLOCK_64K_SIZE) && offset_64k_aligned) {
+			flash_npcx_uma_cmd_by_addr(dev, SPI_NOR_CMD_BE, addr);
+			addr += BLOCK_64K_SIZE;
+			size -= BLOCK_64K_SIZE;
+		} else if ((size >= BLOCK_4K_SIZE) && offset_4k_aligned) {
```

Comment:
Thank you for pointing this out. Following CR is a fix for this: https://github.com/zephyrproject-rtos/zephyr/pull/61105. 

For 4k offset alignment check in for loop, for now we don't need that check but if more options gets added or the offset sector alignment check is removed, then it could work as a shield, do you think it's too much of future proofing